File: app/src/main/java/de/p72b/mocklation/service/room/LocationItemDao.java
package de.p72b.mocklation.service.room;

import java.util.List;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;
import io.reactivex.Flowable;
import io.reactivex.Maybe;
import io.reactivex.Single;

@Dao
public interface LocationItemDao {
    @Query("SELECT * FROM locations")
    Maybe<List<LocationItem>> getAll();

    @Query("SELECT * FROM locations WHERE code IN (:locationItemIds)")
    Flowable<List<LocationItem>> loadAllByIds(int[] locationItemIds);

    @Query("SELECT * FROM locations where code = :code")
    Single<LocationItem> findByCode(String code);

    @Query("SELECT * FROM locations where displayed_name = :displayedName")
    Maybe<List<LocationItem>> findByDisplayedName(String displayedName);


    @Insert
    void insertAll(LocationItem... locationItems);

    @Delete
    void delete(LocationItem locationItem);

    @Update
    void updateLocationItems(LocationItem... locationItems);
}

File: app/src/main/java/de/p72b/mocklation/revamp/room/LocationItemDao.kt
package de.p72b.mocklation.revamp.room

import androidx.lifecycle.LiveData
import androidx.room.*
import io.reactivex.Maybe
import io.reactivex.Single

@Dao
interface LocationItemDao {
    @get:Query("SELECT * FROM locations")
    val all: Maybe<List<LocationItem>>

    @get:Query("SELECT * FROM locations")
    val allLiveData: LiveData<List<LocationItem>>

    @Query("SELECT * FROM locations where code = :code")
    fun findByCode(code: String): Single<LocationItem>

    @Query("SELECT * FROM locations where displayed_name = :displayedName")
    fun findByDisplayedName(displayedName: String): Maybe<List<LocationItem>>

    @Insert
    fun insertAll(vararg locationItems: LocationItem)

    @Delete
    fun delete(locationItem: LocationItem)

    @Update
    fun updateLocationItems(vararg locationItems: LocationItem)
}